code/modules/antagonists/changeling/powers/transform.dm

```diff
@@ -142,23 +142,30 @@
 //Change our DNA to that of somebody we've absorbed.
 /datum/action/changeling/transform/sting_action(mob/living/carbon/human/user)
 	var/datum/antagonist/changeling/changeling = user.mind.has_antag_datum(/datum/antagonist/changeling)
-	var/datum/changelingprofile/chosen_prof = changeling.select_dna("Select the target DNA: ", "Target DNA")
+	var/datum/changelingprofile/chosen_prof = changeling.select_dna()
 
 	if(!chosen_prof)
 		return
 	..()
 	changeling_transform(user, chosen_prof)
 	return TRUE
 
-/datum/antagonist/changeling/proc/select_dna(prompt, title)
+/**
+ * Gives a changeling a list of all possible dnas in their profiles to choose from and returns profile containing their chosen dna
+ */
+/datum/antagonist/changeling/proc/select_dna()
 	var/mob/living/carbon/user = owner.current
 	if(!istype(user))
 		return
-	var/list/names = list("Drop Flesh Disguise")
-	for(var/datum/changelingprofile/prof in stored_profiles)
-		names += "[prof.name]"
 
-	var/chosen_name = input(prompt, title, null) as null|anything in sortList(names)
+	var/list/disguises = list("Drop Flesh Disguise" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_drop"))
+	for(var/datum/changelingprofile/current_profile in stored_profiles)
+		var/datum/icon_snapshot/snap = current_profile.profile_snapshot
+		var/image/disguise_image = image(icon = snap.icon, icon_state = snap.icon_state)
+		disguise_image.overlays = snap.overlays
+		disguises += list("[current_profile.name]" = disguise_image)
```

Comment:
Any reason you did this instead of:

```suggestion
		disguises[current_profile.name] = disguise_image
```

...?

Reply:
None at particular.

Reply:
Actually, there might be one reason to do that instead - profile names are just names of the person that has that given DNA. So there could be multiple profiles with the same name, but completely different DNA features, which would warrant new addition to the list.

Reply:
`list += list` merges the two lists together, it doesn't add a new element. If two people have the same name, it'll override their DNA anyway.

Reply:
Do you have any idea for that, or do you think it would be fine with just suggestion above commited and leaving it as it is?